### PR TITLE
My VA - update appointments link

### PR DESCRIPTION
**Pull request by allisonlu**
## Summary
This PR
- updates the **Schedule and manage your appointments** CTA link from **/health-care/schedule-view-va-appointments/appointments** to **/my-health/appointments** in three places: in the Next appointments card (if applicable), in the list of secondary CTA links (if user doesn't have any upcoming appts), or warning alert when API call fails.
- adds feature toggle (created by the Appointments team) to My VA's mock server and unit tests

This work is behind the `va_online_scheduling_breadcrumb_url_update` (i.e. `vaOnlineSchedulingBreadcrumbUrlUpdate`) feature toggle. This will be removed when the Appointments team launches this to prod.

## Related issue(s)
- My VA Ticket: https://github.com/department-of-veterans-affairs/va.gov-team/issues/63401
- Epic: https://github.com/department-of-veterans-affairs/va.gov-team/issues/62258

- Appointments ticket: https://github.com/department-of-veterans-affairs/va.gov-team/issues/61301

## Testing done
- locally, visually
- unit and e2e tests pass

## What areas of the site does it impact?
My VA - Health care

## Acceptance criteria
- [ ] The "**Schedule and manage your appointments**" link is updated in the appointments card, in secondary link list, and in warning alert

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [x] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [x] Did you login to a local build and verify all authenticated routes work as expected with a test user

### CODE REVIEW

**Review by adamwhitlock1 — COMMENTED** [review]
I think the test setups need to be tweaked a bit on this, so commented on that. Let me know if you have questions.

File: src/applications/personalization/dashboard/tests/components/health-care-v2/AppointmentsV2.unit.spec.jsx [hunk 1]
```diff
@@ -1,16 +1,27 @@
...
+
   it('should render without appointments', () => {
     const start = moment.parseZone();
     const startFormatted = start.format('dddd, MMMM Do, YYYY');
     const timeZone = getAppointmentTimezone();
-    const tree = render(<AppointmentsCard />);
+    const tree = renderWithStoreAndRouter(
+      <AppointmentsCard />,
+      initialLinkState,
+    );
```
[the remark this answers is not in the document]
Comment (allisonlu) [reply src/applications/personalization/dashboard/tests/components/health-care-v2/AppointmentsV2.unit.spec.jsx:24]: There wasn't anything in particular that I was planning on testing, and I didn't think I would need to update any tests, but all these unit tests were failing after making changes to the FE (I was getting the [same error that I was getting about two weeks ago](https://dsva.slack.com/archives/C053JC66YHF/p1691432665994329)) so I was thinking the new feature toggle was messing things up. 

Thanks for the `.debug()` tip! I think I'm going to revisit these tests with that (before I started using `renderWithStoreandRouter` and see what's going on

[the remark this answers is not in the document]
Comment (adamwhitlock1) [reply src/applications/personalization/dashboard/tests/components/health-care-v2/AppointmentsV2.unit.spec.jsx:24]: You probably will still need to use `renderWithStoreAndRouter` to get the tests to render the component, but would probably be fine with just calling it with an empty object as the initialState 🤷🏼, that way the redux store is available and the component wont crash the test when rendering is attempted.

Let me know how that goes 🔎

**Review by adamwhitlock1 — APPROVED** [review]
New changes look good!
